### PR TITLE
Add shelly trv window open state reporting

### PR DESCRIPTION
**Pull request by zewelor**
Hi

Added window open state reporting for TRV. Both sensors for window state, and state on the reporting. I've also checked if there is command mqtt topic for open_window_report setting, but looks like not: https://shelly-api-docs.shelly.cloud/gen1/#shelly-trv-mqtt . Also tried mqtt command topic "shellies/shellytrv-XXXXXXXXXX/thermostat/0/command/open_window_report" with value 1 and didn't saw any change / response.

### CODE REVIEW

File: python_scripts/shellies_discovery.py [hunk 1]
```diff
@@ -1401,6 +1409,22 @@
     KEY_NAME: "Firmware",
     KEY_STATE_TOPIC: TOPIC_INFO,
 }
+OPTIONS_SENSOR_WINDOW_OPEN = {
```
Comment (bieniu) [inline python_scripts/shellies_discovery.py:1412]: I understand that the user has a window sensor that reports the state to TRV, what is the profit of duplicating this sensor here?

File: python_scripts/shellies_discovery.py [hunk 2]
```diff
@@ -1401,6 +1409,22 @@
...
+    KEY_DEVICE_CLASS: DEVICE_CLASS_WINDOW,
+    KEY_NAME: "Window open",
+    KEY_STATE_TOPIC: TOPIC_INFO,
+    KEY_VALUE_TEMPLATE: TPL_WINDOW_OPEN,
+}
+
+OPTIONS_SENSOR_OPEN_WINDOW_REPORT = {
+    KEY_ENABLED_BY_DEFAULT: False,
+    KEY_ENTITY_CATEGORY: ENTITY_CATEGORY_DIAGNOSTIC,
+    KEY_NAME: "Open window report",
```
Comment (bieniu) [inline python_scripts/shellies_discovery.py:1424]: IMO `Window status reporting` is a better name. Similarly, you should also change `OPTIONS_SENSOR_OPEN_WINDOW_REPORT` and `TPL_OPEN_WINDOW_REPORT`.

File: python_scripts/shellies_discovery.py [hunk 3]
```diff
@@ -1401,6 +1409,22 @@
...
+    KEY_ENTITY_CATEGORY: ENTITY_CATEGORY_DIAGNOSTIC,
+    KEY_DEVICE_CLASS: DEVICE_CLASS_WINDOW,
+    KEY_NAME: "Window open",
+    KEY_STATE_TOPIC: TOPIC_INFO,
+    KEY_VALUE_TEMPLATE: TPL_WINDOW_OPEN,
+}
+
+OPTIONS_SENSOR_OPEN_WINDOW_REPORT = {
+    KEY_ENABLED_BY_DEFAULT: False,
+    KEY_ENTITY_CATEGORY: ENTITY_CATEGORY_DIAGNOSTIC,
```
Comment (bieniu) [inline python_scripts/shellies_discovery.py:1423]:
```suggestion
    KEY_ENTITY_CATEGORY: ENTITY_CATEGORY_CONFIG,
```

**Review by bieniu — CHANGES_REQUESTED** [review]

**Review by bieniu — COMMENTED** [review]
I think the window status reporting came with some firmware update. So we should also change the value of `MIN_VALVE_FIRMWARE_DATE`.

File: python_scripts/shellies_discovery.py (hunk 1 above)
Comment (zewelor) [reply python_scripts/shellies_discovery.py:1412]: Not sure if I understand. Its the state returned from the TRV itself. For example check if TRV has correct window open state assumed. Its always possible that updating TRV window open state gets desynchronized, for example info about window being closed got lost etc. 

TRV get window state update via rest api, and then closes when window state reporting is on and window state got open status. Then you need to report window closed status, some time later.

File: python_scripts/shellies_discovery.py (hunk 2 above)
Comment (zewelor) [reply python_scripts/shellies_discovery.py:1424]: Ok I took name from TRV settings, will change as suggested.

File: python_scripts/shellies_discovery.py (hunk 1 above)
Comment (bieniu) [reply python_scripts/shellies_discovery.py:1412]: Ok, I understand your point. But I think the name of the sensor is not good. Maybe `Reported window state`?

**Comment by zewelor** [comment]
Thanks for CR, pushed changes as suggested. Used min firmware date from this post ( https://www.facebook.com/groups/ShellyIoTCommunitySupport/posts/4970138543085470/ ). I hope its correct ?

**Review by bieniu — APPROVED** [review]
I pushed two small changes. Looks good, thanks!

**Comment by zewelor** [comment]
Great thanks !

**Pull request merged**